internal/statemachine/classic_test.go

```diff
@@ -2054,6 +2055,118 @@ func TestStateMachine_FailedPopulateClassicRootfsContents(t *testing.T) {
+		{
+			name:                "set default",
+			existingSourcesList: "deb http://ports.ubuntu.com/ubuntu-ports jammy main restricted",
+			customization:       &imagedefinition.Customization{},
+			expectedSourcesList: `deb http://archive.ubuntu.com/ubuntu/ jammy main restricted universe
+`,
+		},
+		{
+			name:                "set less components",
+			existingSourcesList: "deb http://ports.ubuntu.com/ubuntu-ports jammy main restricted",
+			customization: &imagedefinition.Customization{
+				Components: []string{"main"},
+			},
+			expectedSourcesList: `deb http://archive.ubuntu.com/ubuntu/ jammy main
+`,
+		},
+		{
+			name:                "set components and pocket",
+			existingSourcesList: "deb http://ports.ubuntu.com/ubuntu-ports jammy main restricted",
+			customization: &imagedefinition.Customization{
+				Components: []string{"main"},
+				Pocket:     "security",
+			},
+			expectedSourcesList: `deb http://archive.ubuntu.com/ubuntu/ jammy main
+deb http://security.ubuntu.com/ubuntu/ jammy-security main
+`,
+		},
+		{
+			name:                "fail to write sources.list",
+			existingSourcesList: "deb http://ports.ubuntu.com/ubuntu-ports jammy main restricted",
+			customization: &imagedefinition.Customization{
+				Components: []string{"main"},
+				Pocket:     "security",
+			},
+			expectedSourcesList: "deb http://ports.ubuntu.com/ubuntu-ports jammy main restricted",
+			expectedErr:         "unable to open sources.list file",
+			mockFuncs: func() func() {
+				mock := NewOSMock(
+					&osMockConf{
+						OpenFileThreshold: 0,
+					},
+				)
+
+				osOpenFile = mock.OpenFile
+				return func() { osOpenFile = os.OpenFile }
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			asserter := helper.Asserter{T: t}
+			restoreCWD := helper.SaveCWD()
+			defer restoreCWD()
+
+			var stateMachine ClassicStateMachine
+			stateMachine.commonFlags, stateMachine.stateMachineFlags = helper.InitCommonOpts()
+			stateMachine.parent = &stateMachine
+			stateMachine.ImageDef = imagedefinition.ImageDefinition{
+				Architecture:  getHostArch(),
+				Series:        getHostSuite(),
+				Rootfs:        &imagedefinition.Rootfs{},
+				Customization: tc.customization,
+			}
+
+			err := helper.SetDefaults(&stateMachine.ImageDef)
+			asserter.AssertErrNil(err, true)
+
+			err = stateMachine.makeTemporaryDirectories()
+			asserter.AssertErrNil(err, true)
+
+			t.Cleanup(func() { os.RemoveAll(stateMachine.stateMachineFlags.WorkDir) })
+
+			err = os.MkdirAll(filepath.Join(stateMachine.tempDirs.chroot, "etc", "apt"), 0644)
+			asserter.AssertErrNil(err, true)
+
+			sourcesListPath := filepath.Join(stateMachine.tempDirs.chroot, "etc", "apt", "sources.list")
+
+			err = osWriteFile(sourcesListPath, []byte(tc.existingSourcesList), 0644)
+			asserter.AssertErrNil(err, true)
+
+			if tc.mockFuncs != nil {
+				restoreMock := tc.mockFuncs()
+				t.Cleanup(restoreMock)
+			}
+
+			err = stateMachine.customizeSourcesList()
```

Comment:
I am surprised that this doesn't satisfy codecov above for the `TargetPocketList()` coverage.

Reply:
Me too. I suspect this is because this call of the function is outside the image_definition package and the coverage is not registered.